forc-pkg/src/pkg.rs

```diff
@@ -2244,6 +2238,27 @@ pub fn build_with_options(build_options: BuildOptions) -> Result<Compiled> {
     Ok(compiled)
 }
 
+/// Builds a project with given BuildOptions
+pub fn build_with_options(build_options: BuildOptions) -> Result<()> {
```

Comment:
I think we can still return something useful here. E.g. maybe we could do something like:

1. Rename `Compiled` to `BuiltPackage`. The word `Built` probably better corresponds with the `build` command than `Compiled`, and we add the `Package` suffix to clarify that this is a package (and not a workspace).
2. Create a `Built` enum and return it from `build_with_options`, something like:
   ```rust
   pub enum Built {
       Package(BuiltPackage),
       Workspace,
   }
   ```
   Then in the future, we can potentially add a `BuiltWorkspace` type that contains some kind of summary of the workspace build.
3. Return the `Built` from the `forc::ops::forc_build` too.

The advantage here is that, in the case we built a package, we can still pass the built result directly to the user before it is dropped so that they don't have to read it all into memory again from a file.